tests/Integration/ApiPlatform/ProductListingEndpointTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PsApiResourcesTest\Integration\ApiPlatform;
+
+use Tests\Resources\Resetter\LanguageResetter;
+use Tests\Resources\Resetter\ProductResetter;
+use Tests\Resources\ResourceResetter;
+
+class ProductListingEndpointTest extends ApiTestCase
+{
+    protected static int $frenchLangId;
+
+    public static function setUpBeforeClass(): void
+    {
+        parent::setUpBeforeClass();
+        (new ResourceResetter())->backupTestModules();
+        ProductResetter::resetProducts();
+        LanguageResetter::resetLanguages();
+        self::$frenchLangId = self::addLanguageByLocale('fr-FR');
+        self::createApiClient(['product_read']);
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        parent::tearDownAfterClass();
+        ProductResetter::resetProducts();
+        LanguageResetter::resetLanguages();
+        // Reset modules folder that are removed with the FR language
+        (new ResourceResetter())->resetTestModules();
+    }
+
+    public function getProtectedEndpoints(): iterable
+    {
+        yield 'list endpoint' => [
+            'GET',
+            '/api/products',
+        ];
+    }
+
+    public function testListProducts(): void
+    {
+        $bearerToken = $this->getBearerToken([
+            'product_read',
+        ]);
+
+        $response = static::createClient()->request('GET', '/api/products', ['auth_bearer' => $bearerToken]);
+        self::assertResponseStatusCodeSame(200);
+        self::assertCount(19, json_decode($response->getContent())->items);
```

Comment:
The tests here should be improved, you need to validate the content of the returned data, at least the structure of each product, the filters, order and pagination elements as well